transport/grpc/outbound.go

```diff
@@ -231,3 +231,87 @@ func invokeErrorToYARPCError(err error, responseMD metadata.MD) error {
 	}
 	return intyarpcerrors.NewWithNamef(code, name, message)
 }
+
+// CallStream implements transport.StreamOutbound#CallStream.
+func (o *Outbound) CallStream(ctx context.Context, request *transport.StreamRequest) (transport.ClientStream, error) {
+	if _, ok := ctx.Deadline(); !ok {
+		return nil, yarpcerrors.InvalidArgumentErrorf("stream requests require a connection establishment timeout on the passed in context.")
+	}
+	if err := o.once.WaitUntilRunning(ctx); err != nil {
+		return nil, err
+	}
+	start := time.Now()
+
+	return o.stream(ctx, request, start)
+}
+
+func (o *Outbound) stream(
+	ctx context.Context,
+	req *transport.StreamRequest,
+	start time.Time,
+) (_ transport.ClientStream, err error) {
+	if req.Meta == nil {
+		return nil, yarpcerrors.InvalidArgumentErrorf("stream request requires a request metadata")
+	}
+	treq := req.Meta.ToRequest()
+	md, err := transportRequestToMetadata(treq)
+	if err != nil {
+		return nil, err
+	}
+
+	fullMethod, err := procedureNameToFullMethod(req.Meta.Procedure)
+	if err != nil {
+		return nil, err
+	}
+
+	apiPeer, onFinish, err := o.peerChooser.Choose(ctx, treq)
+	defer func() {
+		if onFinish != nil {
+			onFinish(err)
+		}
+	}()
```

Comment:
nit: Move defer below `err != nil` check. The contract for functions that
return callbacks like this is that if `err` was nil, the callback is non-nil.

```go
apiPeer, onFinish, err := ...
if err != nil {
    return nil, err
}
defer func() { onFinish(err) }()